src/System.Buffers.Experimental/System/Buffers/Sequences/SequenceExtensions.cs

```diff
@@ -12,24 +12,38 @@ public static class Sequence
         public static ReadOnlySpan<byte> ToSpan(this ReadOnlySequence<byte> sequence)
         {
             SequencePosition position = sequence.Start;
-            ResizableArray<byte> array = new ResizableArray<byte>(1024);
+            int capacity = 0;
+            while (sequence.TryGet(ref position, out ReadOnlyMemory<byte> buffer))
+                capacity += buffer.Length;
+
+            position = sequence.Start;
+            var count = 0;
+            byte[] array = new byte[capacity];
             while (sequence.TryGet(ref position, out ReadOnlyMemory<byte> buffer))
             {
-                array.AddAll(buffer.Span);
+                buffer.Span.CopyTo(array.AsSpan(count));
+                count += buffer.Length;
             }
-            return array.Span;
+            return array.AsSpan();
         }
 
         public static ReadOnlySpan<byte> ToSpan<T>(this T sequence) where T : ISequence<ReadOnlyMemory<byte>>
         {
             SequencePosition position = sequence.Start;
-            ResizableArray<byte> array = new ResizableArray<byte>(1024);
+            int capacity = 0;
+            while (sequence.TryGet(ref position, out ReadOnlyMemory<byte> buffer))
+                capacity += buffer.Length;
+
+            position = sequence.Start;
+            var count = 0;
```

Comment:
nit: don't use var here
```suggestion
            int count = 0;
```